api/filters/filtersutil/setters.go

```diff
@@ -14,6 +14,23 @@ func SetScalar(value string) SetFn {
 	}
 }
 
+// SetString returns a SetFn to set a string value
+func SetString(value string) SetFn {
+	n := &yaml.Node{
+		Kind:  yaml.ScalarNode,
+		Value: value,
+		Tag:   yaml.NodeTagString,
+	}
+	if yaml.IsYaml1_1NonString(n) {
+		n.Style = yaml.DoubleQuotedStyle
```

Comment:
I am not sure this is the best place for this. IMO the best place would be in the yaml code but since we can't control that I suppose we have to pick somewhere else. 

Would it be possible to fold this logic into `yaml.FieldSetter`, so that we still use `filtersutil.SetScalar`? That way, it can be shared by all filters that set string values including prefix, suffix, namespace, and labels.

If we do that, it is possible we will have to slightly alter the implementation of `replicacount.set` - which uses `SetScalar` to set an integer value `, but IMO it is a worthwhile tradeoff. 

We also should add a `// TODO` comment to the logic here to remind us that we would eventually like to move the work of quoting nonstring values in string fields to the upstream yaml library.

Reply:
I'll look into it and thanks for reviewing so quickly! I wasn't sure this was the right place to fix this and hoped opening a PR would mean I get feedback like this.
Having `SetScalar` handle this automatically seems ideal since it seems like it's expected that every Node should have a type tag.
I can add a `// TODO`.

Reply:
Thank you, I think it should be possible to put this code in `FieldSetter.Filter` under some if statement `if s.StringValue != ""`, but please report back if you discover that this is more difficult than expected.

Reply:
Since the main issue is that the tag doesn't get set to `yaml.NodeStringTag`, but `Node.StringValue` is used for various value types, to move the check over to kyaml then I think it makes sense to pass 'tag' like for `SetEntry`. Also that  means SetScalar and SetEntry are consistent.
Without setting the tag, checking `s.StringValue != ""` in  `FieldSetter.Filter` and setting to style to `DoubleQuotedStyle` doesn't work and I don't think I can assume that tag type anywhere but where `filtersutil.SetScalar` is called.
It doesn't seem like `fieldspec.CreateTag` needs to be set in the filters packages if this is explicitly set when calling `filtersutil.SetScalar`, and I think this should be handled by kyaml anyway.
It would be nice still if `filtersutil.SetScalar` didn't need to be passed the tag arg, but that might require more changes like adding `filtersutil` funcs or `yaml.Node` fields per tag type. Otherwise there could be a special case for replicacount.
I think I also need to make the 'nonstring' check only check for int type since this is double quoting bools as well.